src/shell/shell.cpp

```diff
@@ -475,7 +475,8 @@ class AUTOEXEC:public Module_base {
 					if(!name) continue;
 				}
 				*name++ = 0;
-				if (access(buffer,F_OK)) continue;
+				if (!path_exists(buffer))
+					continue;
```

Comment:
Another potential gotcha (which now is made clearer).

Gotta love C function returning 0 for success that evaluate to false booleans!

Reply:
I have no problem with C functions returning 0, but in this case original code was bad; it should've been:
```C++
if (access(buffer,F_OK) == -1)
        continue;
```
or
```C++
if (access(buffer,F_OK) != 0)
        continue;
```
To indicate to reader what the code is supposed to do…

(According to documentation, this function can return only 0 or -1)